allennlp/data/tokenizers/word_tokenizer.py

```diff
@@ -38,42 +55,272 @@ class WordTokenizer(Tokenizer):
         # We reverse the tokens here because we're going to insert them with `insert(0)` later;
         # this makes sure they show up in the right order.
         self._start_tokens.reverse()
         self._end_tokens = end_tokens or []
 
+    def _sanitize(self, tokens: List[spacy.tokens.Token]) -> List[Token]:
+        """
+        Converts spaCy tokens to allennlp tokens. Is a no-op if
+        keep_spacy_tokens is True
+        """
+        if not self._keep_spacy_tokens:
+            tokens = [
+                Token(
+                    token.text,
+                    token.idx,
+                    token.lemma_,
+                    token.pos_,
+                    token.tag_,
+                    token.dep_,
+                    token.ent_type_,
+                )
+                for token in tokens
+            ]
+        for start_token in self._start_tokens:
+            tokens.insert(0, Token(start_token, 0))
+        for end_token in self._end_tokens:
+            tokens.append(Token(end_token, -1))
+        return tokens
+
+    @overrides
+    def batch_tokenize(self, texts: List[str]) -> List[List[Token]]:
+        return [
+            self._sanitize(_remove_spaces(tokens))
+            for tokens in self.spacy.pipe(texts, n_threads=-1)
+        ]
+
+    @overrides
+    def tokenize(self, text: str) -> List[Token]:
+        # This works because our Token class matches spacy's.
+        return self._sanitize(_remove_spaces(self.spacy(text)))
+
+    @classmethod
+    def from_params(cls, params: Params) -> "SpacyWordTokenizer":  # type: ignore
+
+        # Backwards compatibility for legacy "word" Tokenizer
+        # which provided constructor arguments inside "word_splitter" key.
+        word_splitter_params = params.get("word_splitter")
+        if isinstance(word_splitter_params, Params) and word_splitter_params.pop("type") == "spacy":
+            params = word_splitter_params
+        elif isinstance(word_splitter_params, str) and word_splitter_params == "spacy":
+            params = Params({})  # Empty
+
+        language = params.pop("language", "en_core_web_sm")
+        pos_tags = params.pop_bool("pos_tags", False)
+        parse = params.pop_bool("parse", False)
+        ner = params.pop_bool("ner", False)
+        keep_spacy_tokens = params.pop_bool("keep_spacy_tokens", False)
+        split_on_spaces = params.pop_bool("split_on_spaces", False)
+        start_tokens = params.pop("start_tokens", None)
+        end_tokens = params.pop("end_tokens", None)
+
+        params.assert_empty(cls.__name__)
+
+        return cls(
+            language=language,
+            pos_tags=pos_tags,
+            parse=parse,
+            ner=ner,
+            keep_spacy_tokens=keep_spacy_tokens,
+            split_on_spaces=split_on_spaces,
+            start_tokens=start_tokens,
+            end_tokens=end_tokens,
+        )
+
+
+@Tokenizer.register("simple")
+class SimpleSpacyWordTokenizer(Tokenizer):
```

Comment:
Should just be `SimpleTokenizer`.  Probably also ok to just remove it.  You should use spacy or bling instead.